hw/top_earlgrey/top_earlgrey.core

```diff
@@ -24,8 +24,10 @@ filesets:
       - lowrisc:constants:top_pkg
       - lowrisc:ip:nmi_gen
     files:
-      - ip/xbar/rtl/autogen/tl_main_pkg.sv
-      - ip/xbar/rtl/autogen/xbar_main.sv
+      - ip/xbar_main/rtl/autogen/tl_main_pkg.sv
```

Comment:
just a thought, in the future, should we auto-generate the xbar related core files? For example, if we imagine someone coming in a re-organizing the bus hierarchy and let's say added another peri, or removed peri, it would be nice to not have to modify top_earlgrey.core every time, but instead just declare a dependency on some generated core files.

Reply:
That was also in my mind as well. But it could create conflict if multiple tops exist and the tops have common xbar name (eg. xbar_main).

Reply:
To solve this, we can use multiple cores-root in fusesoc but that means we shold move top_name to upper directory. (`fusesoc --cores-root hw/ --cores-root top/top_earlgrey` something like this)

Reply:
you mean multiple tops like top_fpga, top_verilator etc? Is there any reason to think their xbar_mains would be different though?

Reply:
o if you're referring to tops like different projects, then i completely agree.  I think if we ever want to maintain multiple distinct tops, having different core.roots might be inevitable... For now though, I think it might be a reasonable step to generate those core files, and try to massage the larger picture later.  

Anyhow this was just a suggestion, in the short term, I doubt we will have a ton of movement on the bus structure.

Reply:
The latter is what I meant.

Reply:
yeah i think for different project tops, this is inevitable, and I think links with the larger story of exactly how people will take this repo and go forward.  I would argue for now, let's assume we have the one and go from there. In the future, if we need to massage it more, I'm sure we will find a way.